source/common/access_log/access_log_formatter.cc

```diff
@@ -87,21 +105,26 @@ void AccessLogFormatParser::parseCommand(const std::string& token, const size_t
     max_length = length_value;
   }
 
-  std::string header_name = token.substr(start, end_request - start);
-  size_t separator = header_name.find('?');
+  std::string name_data = token.substr(start, end_request - start);
+  size_t separator_pos = name_data.find(separator);
 
-  if (separator == std::string::npos) {
-    main_header = header_name;
-    alternative_header = "";
+  if (separator_pos == std::string::npos) {
+    main = name_data;
   } else {
-    main_header = header_name.substr(0, separator);
-    alternative_header = header_name.substr(separator + 1, end_request - separator - 1);
+    main = name_data.substr(0, separator_pos);
```

Comment:
Can you add comments/examples of what is going on here, it's a bit hard to process without context.

Reply:
Will do